pkg/image/apis/image/helper.go

```diff
@@ -523,6 +539,44 @@ func ImageWithMetadata(image *Image) error {
 	return nil
 }
 
+// FetchImageSignature attemps to fetch the signature blob for the provided
+// image and mutates the image object with the signature content if the registry
+// has defined a known signature store.
+func FetchImageSignature(registry string, image *Image) error {
```

Comment:
Besides HTTPS sigstore, this should perhaps also support `file` sigstores (for signatures on a local NFS), and _definitely_ the `X-Registry-Supports-Signatures` protocol (which does not require any client-side configuration), in addition to this one.

Are you _sure_ this can’t reuse `github.com/containers/image/docker` for all of that?